test/e2e/runtime_kappnav.go

```diff
@@ -0,0 +1,209 @@
+	}
+
+	if err = useExistingApplications(t, f, ctx); err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+}
+
+func createKappNavApplication(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	ns, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	const name string = "example-runtime-kappnav"
+
+	runtime := util.MakeBasicRuntimeComponent(t, f, name, ns, 1)
+	runtime.Spec.ApplicationName = appName
+
+	err = f.Client.Create(goctx.TODO(), runtime, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	if err != nil {
+		return err
+	}
+
+	// Verify readiness of created resource
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, ns, name, 1, retryInterval, timeout)
+	if err != nil {
+		return err
+	}
+
+	target := types.NamespacedName{Namespace: ns, Name: name}
+
+	ok, err := verifyKappNavLabels(t, f, target)
+	if err != nil {
+		return err
+	} else if !ok {
+		return errors.New("could not find kappnav labels")
+	}
+	t.Log("kappnav labels found")
+
+	err = util.WaitForApplicationCreated(t, f, types.NamespacedName{Name: appName, Namespace: ns})
+	if err != nil {
+		return err
+	}
+	t.Log("related application definition found")
+
+	return nil
+}
+
+func updateKappNavApplications(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	ns, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	const name string = "example-runtime-kappnav"
+
+	target := types.NamespacedName{Namespace: ns, Name: name}
+
+	err = util.UpdateApplication(f, target, func(r *appstacksv1beta1.RuntimeComponent) {
+		appDef := false
+		r.Spec.CreateAppDefinition = &appDef
+	})
+	if err != nil {
+		return err
+	}
+
+	ok, err := verifyKappNavLabels(t, f, target)
+	if err != nil {
+		return err
+	} else if !ok {
+		return errors.New("kappnav labels present after disabling")
+	}
+	t.Log("kappnav labels successfully removed")
+
+	err = util.WaitForApplicationDelete(t, f, types.NamespacedName{Name: appName, Namespace: ns})
+	if err != nil {
+		return err
+	}
+	t.Log("created application definition removed")
+
+	return nil
+}
+
+func useExistingApplications(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	ns, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	const name string = "example-runtime-kappnav"
+	var existingAppName string = "existing-app"
+	// Add selector labels to verify that app was actually found
+	selectMatchLabels := map[string]string{
+		"test-key": "test-value",
+	}
+
+	// create existing application
+	err = util.CreateApplicationTarget(f, ctx, types.NamespacedName{Name: existingAppName, Namespace: ns}, selectMatchLabels)
+	if err != nil {
+		return err
+	}
+
+	// connect to existing application IN namespace
+	target := types.NamespacedName{Namespace: ns, Name: name}
+
+	err = util.UpdateApplication(f, target, func(r *appstacksv1beta1.RuntimeComponent) {
+		createApp := false
+		r.Spec.CreateAppDefinition = &createApp
```

Comment:
Operator should find the Application CR matching the application name in the current namespace, even if CreateAppDefinition is not explicitly set to false. So this line should be removed.